File: examples/DropWizardExtraJava/src/main/java/com/instil/errors/CoursesExceptionMapper.java
package com.instil.errors;

import static javax.ws.rs.core.Response.Status.SERVICE_UNAVAILABLE;

import javax.ws.rs.ext.ExceptionMapper;
import javax.ws.rs.core.Response;

public class CoursesExceptionMapper implements ExceptionMapper<NoCoursesException> {
    @Override
    public Response toResponse(NoCoursesException ex) {
        String msg = String.format("[\"%s\"]", ex.getMessage());
        return Response.status(SERVICE_UNAVAILABLE)
                .entity(msg)
                .build();
    }
}

File: examples/DropWizardExtraKotlin/src/main/kotlin/com/instil/errors/CoursesExceptionMapper.kt
package com.instil.errors

import javax.ws.rs.core.Response
import javax.ws.rs.ext.ExceptionMapper

class CoursesExceptionMapper : ExceptionMapper<NoCoursesException> {
    override fun toResponse(ex: NoCoursesException): Response {
        return Response.status(Response.Status.SERVICE_UNAVAILABLE)
            .entity("[\"${ex.message}\"]")
            .build()
    }
}
